CoroutineTemplate/app/src/test/java/co/nimblehq/coroutine/ui/BaseFragmentTest.kt

```diff
@@ -0,0 +1,69 @@
+package co.nimblehq.coroutine.ui
+
+import android.content.ComponentName
+import android.content.Intent
+import android.os.Bundle
+import androidx.annotation.StyleRes
+import androidx.core.util.Preconditions
+import androidx.test.core.app.ActivityScenario
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.viewbinding.ViewBinding
+import co.nimblehq.coroutine.EmptyHiltActivity
+import co.nimblehq.coroutine.R
+import co.nimblehq.coroutine.ui.base.BaseFragment
+import dagger.hilt.android.testing.*
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+abstract class BaseFragmentTest<F : BaseFragment<VB>, VB : ViewBinding> {
+
+    protected lateinit var fragment: F
+
+    /**
+     * Issue: launchFragmentInContainer from the androidx.fragment:fragment-testing library
+     * is NOT possible to use right now as it uses a hardcoded Activity under the hood
+     * (i.e. [EmptyFragmentActivity]) which is not annotated with @AndroidEntryPoint.
+     *
+     * As a workaround, use this function that is equivalent. It requires you to add
+     * [HiltTestActivity] in the debug folder and include it in the debug AndroidManifest.xml file
+     * as can be found in this project.
+     * Read more: https://dagger.dev/hilt/testing
+     * Solution: launchFragmentInHiltContainer is copied from architecture-sample repo, but has
+     * been modified to receive input [onInstantiate] so the devs can create the fragment
+     * instance with mock/ fake variables at the initial step.
+     * Reference: https://github.com/android/architecture-samples/blob/dev-hilt/app/src/androidTest/java/com/example/android/architecture/blueprints/todoapp/HiltExt.kt
+     */
```

Comment:
@luongvo 👋

Reply:
@doannimble The thing is [`androidx.fragment:fragment-testing` needs to be imported via `debugImplementation`](https://developer.android.com/guide/fragments/test#declare-dependencies) so the custom `EmptyHiltActivity` needs to be added into at least `debug` flavor, not `test` flavor.